packages/react-core/src/demos/DataList/examples/DataListPagination.tsx

```diff
@@ -0,0 +1,187 @@
+
+import CodeBranchIcon from '@patternfly/react-icons/dist/esm/icons/code-branch-icon';
+import CodeIcon from '@patternfly/react-icons/dist/esm/icons/code-icon';
+import CubeIcon from '@patternfly/react-icons/dist/esm/icons/cube-icon';
+import { rows } from '@patternfly/react-core/dist/esm/demos/sampleData';
+
+export const DataListPagination: React.FunctionComponent = () => {
+  const [page, setPage] = React.useState<number | undefined>(1);
+  const [perPage, setPerPage] = React.useState<number>(10);
+  const [paginatedRows, setPaginatedRows] = React.useState(rows.slice(0, 10));
+
+  const handleSetPage = (
+    _evt: React.MouseEvent | React.KeyboardEvent | MouseEvent,
+    newPage: number,
+    _perPage: number | undefined,
+    startIdx: number | undefined,
+    endIdx: number | undefined
+  ) => {
+    setPaginatedRows(rows?.slice(startIdx, endIdx));
+    setPage(newPage);
+  };
+
+  const handlePerPageSelect = (
+    _evt: React.MouseEvent | React.KeyboardEvent | MouseEvent,
+    newPerPage: number,
+    newPage: number | undefined,
+    startIdx: number | undefined,
+    endIdx: number | undefined
+  ) => {
+    setPaginatedRows(rows.slice(startIdx, endIdx));
+    setPage(newPage);
+    setPerPage(newPerPage);
+  };
+
+  const renderPagination = (id: string, variant: PaginationVariant, isCompact: boolean, isSticky: boolean) => (
+    <Pagination
+      id={id}
+      variant={variant}
+      itemCount={rows.length}
+      page={page}
+      perPage={perPage}
+      isCompact={isCompact}
+      isSticky={isSticky}
+      onSetPage={handleSetPage}
+      onPerPageSelect={handlePerPageSelect}
+      titles={{
+        paginationAriaLabel: `${variant} pagination`
+      }}
+    />
+  );
+
+  const toolbarItems = (
+    <React.Fragment>
+      <ToolbarItem variant="bulk-select">
+        <MenuToggle
+          aria-label="Select data list items"
+          splitButtonOptions={{
+            items: [
+              <MenuToggleCheckbox
+                id="split-dropdown-checkbox"
+                key="split-dropdown-checkbox"
+                aria-label={'Select all data list items'}
+              />
+            ]
+          }}
+        ></MenuToggle>
+      </ToolbarItem>
+      <ToolbarItem>
+        <MenuToggle>Filter by creator name</MenuToggle>
+      </ToolbarItem>
+      <ToolbarItem variant="overflow-menu">
+        <OverflowMenu breakpoint="md">
+          <OverflowMenuItem>
+            <Button variant="primary">Create instance</Button>
+          </OverflowMenuItem>
+          <OverflowMenuControl hasAdditionalOptions>
+            <MenuToggle aria-label="Toolbar kebab overflow menu" variant="plain">
+              <EllipsisVIcon />
+            </MenuToggle>
+          </OverflowMenuControl>
+        </OverflowMenu>
+      </ToolbarItem>
+      <ToolbarItem variant="pagination" align={{ default: 'alignRight' }}>
+        {renderPagination('top-pagination', PaginationVariant.top, false, false)}
```

Comment:
To match core (and the basic [react data-list demo](https://www.patternfly.org/components/data-list/react-demos/basic/)) this would use compact top pagination

```suggestion
        {renderPagination('top-pagination', PaginationVariant.top, true, false)}
```